packages/okta-auth-js/lib/token.js

```diff
@@ -486,49 +497,51 @@ function getToken(sdk, oauthOptions, options) {
           var windowOrigin = getOrigin(sdk.idToken.authorize._getLocationHref());
           var redirectUriOrigin = getOrigin(oauthParams.redirectUri);
           if (windowOrigin !== redirectUriOrigin) {
-            return Q.reject(new AuthSdkError('Using fragment, the redirectUri origin (' + redirectUriOrigin +
-              ') must match the origin of this page (' + windowOrigin + ')'));
+            throw new AuthSdkError('Using fragment, the redirectUri origin (' + redirectUriOrigin +
+              ') must match the origin of this page (' + windowOrigin + ')');
           }
           popupPromise = addFragmentListener(sdk, windowEl, options.timeout);
         }
 
         // Both postMessage and fragment require a poll to see if the popup closed
-        var popupDeferred = Q.defer();
-        /* eslint-disable-next-line no-case-declarations, no-inner-declarations */
-        function hasClosed(win) {
-          if (!win || win.closed) {
-            popupDeferred.reject(new AuthSdkError('Unable to parse OAuth flow response'));
-            return true;
-          }
-        }
-        var closePoller = setInterval(function() {
-          if (hasClosed(windowEl)) {
-            clearInterval(closePoller);
+        var closePoller;
+        var popupClosedPromise = new Promise(function(resolve, reject) {
+          /* eslint-disable-next-line no-case-declarations, no-inner-declarations */
+          function hasClosed(win) {
```

Comment:
Function name implies a boolean return w/o side effects, but it looks to actually to be all about a side-effect.  Perhaps pull the `reject()` out and call `reject()` explicitly where `hasClosed()` is called?  (At which point you can define `hasClosed()` externally and don't need to override eslint)

Reply:
I will refactor this section a bit